src/wallet/mod.rs

```diff
@@ -652,21 +669,36 @@ where
         details.sent = selected_amount;
 
         let mut fee_amount = fee_amount.ceil() as u64;
+        if builder.has_absolute_fee() {
+            fee_amount = match builder.fee_policy.as_ref().unwrap() {
+                FeePolicy::FeeAmount(amount) => *amount,
+                _ => fee_amount,
+            }
+        };
         let removed_output_fee_cost = (serialize(&removed_updatable_output).len() as f32
             * new_feerate.as_sat_vb())
         .ceil() as u64;
 
         let change_val = selected_amount - amount_needed - fee_amount;
         let change_val_after_add = change_val.saturating_sub(removed_output_fee_cost);
         if !builder.send_all && !change_val_after_add.is_dust() {
-            removed_updatable_output.value = change_val_after_add;
-            fee_amount += removed_output_fee_cost;
-            details.received += change_val_after_add;
+            if builder.has_absolute_fee() {
+                removed_updatable_output.value = change_val_after_add + removed_output_fee_cost;
+                details.received += change_val_after_add + removed_output_fee_cost;
+            } else {
+                removed_updatable_output.value = change_val_after_add;
+                fee_amount += removed_output_fee_cost;
+                details.received += change_val_after_add;
+            }
 
             tx.output.push(removed_updatable_output);
         } else if builder.send_all && !change_val_after_add.is_dust() {
-            removed_updatable_output.value = change_val_after_add;
-            fee_amount += removed_output_fee_cost;
+            if builder.has_absolute_fee() {
+                removed_updatable_output.value = change_val_after_add + removed_output_fee_cost;
+            } else {
+                removed_updatable_output.value = change_val_after_add;
+                fee_amount += removed_output_fee_cost;
+            }
```

Comment:
Same as above, if the `fee_rate` is correctly set to `0.0` you shouldn't need to handle the absolute fees as a special case, and you should be able to just leave the code as it was before.